pages/article/[pid].js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+import { useRouter } from 'next/router';
+import { collection, onSnapshot, query, where } from 'firebase/firestore';
+import Link from 'next/link';
+
+import { db } from '../../firebase/initFirebase';
+
+const Article = () => {
+  const router = useRouter();
+  const { pid } = router.query;
+
+  const [article, setArticle] = useState([]);
+
+  useEffect(() => {
+    if (router.isReady) {
+      const articlesCollectionRef = collection(db, 'articles');
+      const q = query(articlesCollectionRef, where('link', '==', pid));
+
+      onSnapshot(q, (snapshot) => {
+        setArticle(
+          snapshot.docs.map((doc) => ({
+            ...doc.data(),
+            id: doc.id,
+          })),
+        );
+      });
```

Comment:
I do recommend abstracting this to a helper function like so - https://github.com/GLObus303/showcase-project/blob/master/src/utils/apiTransform.ts#L7-L15

Since you will use this many times